### DISCUSSION TITLE
Many parser errors and warnings do not show a source position

### DISCUSSION THREAD

**Discussion by ntrel** [vlang/v#11731] [Ideas]
There are a lot of uses of p.error() or p.warn() that don't show a position.

#6997 shows how to fix these. Probably best to fix them in batches.

**Comment by ItsaMeTuni** [comment]
I took a look at the code and it seems that  `error` and `warn` are basically just wrappers to `error_with_pos` and `warn_with_pos`, respectively. So the problem isn't that positions aren't shown in some errors, but rather some errors don't show the positions they should, right?

e.g. the error for `$test()` will only highlight the first paren:
```
main.v:3:7: error: only `$tmpl()`, `$env()`, `$embed_file()` and `$vweb.html()` comptime functions are supported right now
    1 | 
    2 | fn main() {
    3 |     $test()
      |          ^
    4 | }
```

I'm currently trying to fix this, but some guidance would be very helpful, as there's very little documentation on the compiler's code. In `comptime.comp_call`, I'm trying to get the whole call to be highlighted in the error, like so:

```
main.v:3:2: error: only `$tmpl()`, `$env()`, `$embed_file()` and `$vweb.html()` comptime functions are supported right now
    1 | 
    2 | fn main() {
    3 |     $test()
      |     ~~~~~~~
    4 | }
```


`$test()` gets hihglighted all the way from `$` to `)`, so far so good, I've been able to accomplish this using the `extend` method on `Position`. However, if there are any parameters like `$test("a", "b")`, only `$test("a"` will get highlighted because I'm limited to using `peek_tok`, `peek_tok2` and `peek_tok3`.

```
main.v:3:2: error: only `$tmpl()`, `$env()`, `$embed_file()` and `$vweb.html()` comptime functions are supported right now
    1 | 
    2 | fn main() {
    3 |     $test("a", "b")
      |     ~~~~~~~~~
    4 | }
```

How can I fix this? I was thinking of using `p.check` calls to check if the call is well-formed before checking if it's a supported comptime function. This way we catch syntax errors early on and can highlight the whole comptime function call in the error.

For example, this is the current output for `$test("a", "b"`
```
main.v:3:7: error: only `$tmpl()`, `$env()`, `$embed_file()` and `$vweb.html()` comptime functions are supported right now
    1 | 
    2 | fn main() {
    3 |     $test("a", "b"
      |          ^
    4 | }
```
The compiler is more concerned with the fact that `test` is not a supported comptime function than the missing parens at the end of the line. Also, shouldn't the check for supported comptime calls happen in the checker rather than the parser?

 Another option is to just highlight `$test` (or even just `test`), but I'm not sure if it's the best way to go.

Thanks for your help!
